modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js

```diff
@@ -133,3 +125,83 @@ function Row({children, item}) {
 		rowContent
 	);
 }
+
+function Column({children, item}) {
+	// eslint-disable-next-line
+	const [{canDrop, isDragging}, drop] = useDrop({
+		accept: [LAYOUT_DATA_ITEM_TYPES.fragment]
+	});
+
+	const {size} = item.config;
+
+	return (
+		<div className={classNames('col', {[`col-${size}`]: size})} ref={drop}>
+			{children}
+		</div>
+	);
+}
+
+function Fragment({item}) {
+	const {fragmentEntryLinks} = useContext(StoreContext);
+
+	const fragmentEntryLink =
+		fragmentEntryLinks[item.config.fragmentEntryLinkId];
+
+	let markup = '';
+
+	if (typeof fragmentEntryLink.content === 'string') {
+		markup = fragmentEntryLink.content;
+	} else if (fragmentEntryLink.content.value && fragmentEntryLink.content.value.content) {
+		markup = fragmentEntryLink.content.value.content;
+	} else {
+		markup = `<div>No markup from ${item.config.fragmentEntryLinkId}</div>`;
+	}
+
+	return (
+		<UnsafeHTML
+			className="fragment"
+			markup={markup}
+		/>
+	);
+}
+
+const LAYOUT_DATA_ITEMS = {
+	[LAYOUT_DATA_ITEM_TYPES.column]: Column,
+	[LAYOUT_DATA_ITEM_TYPES.container]: Container,
+	[LAYOUT_DATA_ITEM_TYPES.fragment]: Fragment,
+	[LAYOUT_DATA_ITEM_TYPES.root]: Root,
+	[LAYOUT_DATA_ITEM_TYPES.row]: Row
+};
+
+const LayoutDataItem = React.forwardRef(({item, layoutData}, ref) => {
+	const Component = LAYOUT_DATA_ITEMS[item.type];
+
+	return (
+		<Component item={item}>
+			{item.children.map(childId => {
+				return (
+					<LayoutDataItem
+						drag={ref}
+						item={layoutData.items[childId]}
+						key={childId}
+						layoutData={layoutData}
+					/>
+				);
+			})}
+		</Component>
+	);
+});
+
+export default function PageEditor() {
+	const mainItem = layoutData.items[layoutData.rootItems.main];
+
+	const isMounted = useIsMounted();
+
+	useEffect(() => {
+		if (isMounted()) {
+			// TODO: send updated layoutData to backend
```

Comment:
I'm a little curious because we're going to update the data here, I am assuming that we will not always update data when we `dispatch` to `store`.

Reply:
That's the part I also am not sure about.

Basically the intent was to send the updated `layoutData` object each time it changes, to keep it in sync with the backend. This `layoutData` object is fetched but the store and used to render  the various "LayoutData" items (container, column, row, fragment) 

I'm not sure where to do that yet, but we can talk about it.

Reply:
Hmm, I see I think this will depend on how the server needs to know the information, if we need to send only the fragments that have been moved or the whole `layoutData`... I think depending on the case decide if we will do it here or whenever we update a change in the store, for example move, edit etc...